crates/viewer/re_space_view_spatial/src/visualizers/assets3d.rs

```diff
@@ -144,38 +146,61 @@ impl VisualizerSystem for Asset3DVisualizer {
 
         let mut instances = Vec::new();
 
-        super::entity_iterator::process_archetype::<Self, Asset3D, _>(
+        super::entity_iterator::process_archetype2::<Self, Asset3D, _>(
             ctx,
             view_query,
             context_systems,
             |ctx, spatial_ctx, results| {
-                use re_space_view::RangeResultsExt as _;
-
-                let resolver = ctx.recording().resolver();
+                use re_space_view::RangeResultsExt2 as _;
 
-                let blobs = match results.get_required_component_dense::<Blob>(resolver) {
-                    Some(blobs) => blobs?,
-                    _ => return Ok(()),
+                let Some(all_blob_chunks) = results.get_required_chunks(&Blob::name()) else {
+                    return Ok(());
                 };
 
-                let media_types = results.get_or_empty_dense(resolver)?;
-                let transforms = results.get_or_empty_dense(resolver)?;
+                let timeline = ctx.query.timeline();
+                let all_blobs_indexed = all_blob_chunks.iter().flat_map(|chunk| {
+                    itertools::izip!(
+                        chunk.iter_component_indices(&timeline, &Blob::name()),
+                        chunk.iter_buffer::<u8>(&Blob::name())
+                    )
+                });
+                let all_media_types = results.iter_as(timeline, MediaType::name());
+
+                // TODO(#6831): we have to deserialize here because `OutOfTreeTransform3D` is
+                // still a complex type at this point.
+                let all_transform_chunks =
+                    results.get_optional_chunks(&OutOfTreeTransform3D::name());
+                let mut all_transform_iters = all_transform_chunks
+                    .iter()
+                    .map(|chunk| chunk.iter_component::<OutOfTreeTransform3D>())
+                    .collect_vec();
+                let all_transforms_indexed = {
+                    let all_albedo_textures =
```

Comment:
wrong name. We're not dealing with albedo textures here, but transforms

Reply:
:( LSP rename betrayal, a story as old as time